spanner/client.go

```diff
@@ -186,6 +187,10 @@ type ClientConfig struct {
 
 	// BatchTimeout specifies the timeout for a batch of sessions managed sessionClient.
 	BatchTimeout time.Duration
+
+	// UseNumber causes the Decoder to unmarshal a number into an interface{} as a
+	// Number instead of as a float64.
```

Comment:
I'm only able to understand this comment because I know the context of this PR. Without that context, this comment contains too little information. Can we add:
1. Additional context that explains that this is for the JSON decoder. (`the Decoder` is not a clear reference to JSON decoding)
2. What it means that something is being unmarshalled as a `Number` instead of a `float64`? So what are the pros/cons of each?

So for example:

UseNumber specifies whether number values inside a Cloud Spanner JSON value should be decoded as a Number or a float64. Decoding to a Number guarantees that the precision used by Cloud Spanner is preserved. Decoding to a float64 can cause loss of precision. The default JSON decode function in Go uses float64. This is therefore also the default used by this client library. Change this value to `true` to prevent loss of precision.